packages/server/src/fhir/repo.ts

```diff
@@ -1282,24 +1282,38 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
       validator.isUUID(resolveId(resource.project) ?? '')
     ) {
       // Deprecated - to be removed after migrating all tables to use "projectId" column
-      result.push(resource.project);
+      compartments.add(resource.project.reference);
     }
 
     if (resource.meta?.account) {
       const id = resolveId(resource.meta.account);
       if (id && validator.isUUID(id)) {
-        result.push(resource.meta.account);
+        compartments.add(resource.meta.account.reference as string);
       }
     }
 
     for (const patient of getPatients(resource)) {
       const patientId = resolveId(patient);
       if (patientId && validator.isUUID(patientId)) {
-        result.push(patient);
+        compartments.add(patient.reference);
       }
     }
 
-    return result;
+    // Carry forward anything added to the resource compartments array
+    if (resource.meta?.compartment?.length) {
+      for (const compartment of resource.meta.compartment) {
```

Comment:
We had discussed having a max compartments length. What should the behavior be if the user specifies more than that max?

Reply:
Good point.  I'd vote for HTTP 400 if greater than N, where N should probably be a server config setting with a default of ~10.

Reply:
> What should the behavior be if the user specifies more than that max?

It would be a performance hit, loading up the `compartments` array.  Not great, but not the end of the world.